src/plugins/intel_gpu/src/plugin/transformations/convert_fc_to_compressed.cpp

```diff
@@ -97,9 +97,9 @@ ConvertFullyConnectedToFullyConnectedCompressed::ConvertFullyConnectedToFullyCon
             return std::make_shared<ov::op::v0::Constant>(*constant, new_shape);
         };
 
-        auto convert_u4const_to_u8 = [convert_u4zp_to_u8](std::shared_ptr<ov::Node> node) {
+        auto convert_u4const_to_u8 = [convert_zp_to_u8](std::shared_ptr<ov::Node> node) {
             auto constant = std::dynamic_pointer_cast<ov::op::v0::Constant>(node);
-            if (constant->get_element_type() != ov::element::u4 || !convert_u4zp_to_u8)
+            if (constant->get_element_type() == ov::element::u8 || !convert_zp_to_u8)
                 return std::dynamic_pointer_cast<ov::Node>(constant);
             return std::dynamic_pointer_cast<ov::Node>(std::make_shared<ov::op::v0::Convert>(node, ov::element::u8));
         };
```

Comment:
Is this conversion necessary in both iGPU/dGPU case? If so, I think we can remove the condition `convert_u4zp_to_u8` itself.